src/main/java/blusunrize/immersiveengineering/api/crafting/FluidTagInput.java

```diff
@@ -170,4 +173,22 @@ public void write(PacketBuffer out)
 	{
 		return TagCollectionManager.getManager().getFluidTags();
 	}
+
+	public boolean extractFrom(IFluidHandler handler)
+	{
+		int remaining = amount;
+		for(int tank = 0; tank < handler.getTanks(); tank++)
+		{
+			FluidStack inTank = handler.getFluidInTank(tank);
+			if(testIgnoringAmount(inTank))
+			{
+				int extractAmount = Math.min(inTank.getAmount(), remaining);
```

Comment:
Given how the fluid handler API is I am not sure if this line makes sense, because the drain implementation isn't requested from a specific tank but left up to the fluid handler to decide how to implement, there should be no harm in requesting to extract remaining, even if the specific tank you are looking at type wise contains less contents. If more can't be extracted the drain method should not return more than was actually able to be extracted.